autoscaler/controllers/common/processors.go

```diff
@@ -1,45 +1,12 @@
 package common
 
 import (
-	"fmt"
 	"sort"
 
-	"encoding/json"
-
 	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
-	"github.com/odigos-io/odigos/common"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-func IsProcessorTracingEnabled(processor *odigosv1.Processor) bool {
-	for _, signal := range processor.Spec.Signals {
-		if signal == common.TracesObservabilitySignal {
-			return true
-		}
-	}
-	return false
-}
-
-func IsProcessorMetricsEnabled(processor *odigosv1.Processor) bool {
-	for _, signal := range processor.Spec.Signals {
-		if signal == common.MetricsObservabilitySignal {
-			return true
-		}
-	}
-	return false
-}
-
-func IsProcessorLogsEnabled(processor *odigosv1.Processor) bool {
-	for _, signal := range processor.Spec.Signals {
-		if signal == common.LogsObservabilitySignal {
-			return true
-		}
-	}
-	return false
-}
-
 func FilterAndSortProcessorsByOrderHint(processors *odigosv1.ProcessorList, collectorRole odigosv1.CollectorsGroupRole) []*odigosv1.Processor {
```

Comment:
I am thinking, should this function be moved to common as well? we might need to reuse this functionality with other platforms - filtering out disabled processors, sorting by order hint and choosing the right role to use. 

We can also address it later on if we run into the situation

Reply:
I am not sure. We could go that route as well, I guess, but I don't necessary think we need it nor do I think it's better experience:

1. Order hints are useful only if we cannot control the order,
2. In a local file one could argue that removing items from yaml or commenting them out is simpler than having an additional attribute,
3. we have one type of collector for now so filtering by role is not necessary.

So I think the question end up being whether we want to maximally mirror every aspect of the environments or not. The config files will be different from k8s resources anyway so I've been working on an assumption that it didn't matter. What do you think?

Reply:
> 1. Order hints are useful only if we cannot control the order,

Not sure I understood. how will the user hint the order of the processors otherwise?

> * 
> * In a local file one could argue that removing items from yaml or commenting them out is simpler than having an additional attribute,

While yaml files are one way to configure the destinations, there can be other ways like via odigos cloud, where the user has no control over files, and the changes are applied via UI. we can fork to use different scheme and build custom configurations per use-case, but it might be hard to maintain and implement vs aligning everywhere to use the same scheme.

Anyway, we can move forward and address this issue if we run into it.

Reply:
> Not sure I understood. how will the user hint the order of the processors otherwise?

The order is clear whenever we're working with arrays. But I agree, that when the items are separated from one another(like in k8s or perhaps a DB schema where every processor is it's own row), we do need a way to order and change the order.

> there can be other ways like via odigos cloud, where the user has no control over files

I see what you mean now. My default way to design this would've been different from the ground, I guess.

I'll leave it as it is for now. But thanks for very valuable feedback. :pray: